### ISSUE TITLE
CMake generate step failed to find sdl2 library though it was installed

### ISSUE THREAD

**Issue by turlapati** [sr99622/libonvif#51]
me@myUbuntu2204:~/workspace/libonvif/build$ cmake -DBUILD_GUI=ON ..
-- Configuring done
CMake Error at libavio/CMakeLists.txt:65 (target_link_libraries):
  Target "avio" links to:

    SDL2::SDL2

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.

**Comment by sr99622** [comment]
The cmake find_package(SDL2 REQUIRED) command in the libavio/CMakeLists.txt is where the SDL2 package is found.  The REQUIRED flag in the command should throw an error if the package is not found. 

The files cmake is looking for are in the libsdl2-dev package.  You can verify that this is installed using the command 'apt list | grep libsdl2-dev'.  The library file is libSDL2-2.0.so which should be found in the directory /lib/x86_64-linux-gnu.  You should also have directory /usr/include/SDL2 which has the include files.

An easy thing to try would be to uninstall libsdl2-dev and the re-install it see if that fixes the problem.

**Comment by turlapati** [comment]
The issue is similar to one discussed at: https://discourse.libsdl.org/t/how-is-sdl2-supposed-to-be-used-with-cmake/31275/5. I tried all the suggestions and they didn't work

I am using Ubuntu 20.04 and the official version of SDL is 2.0.10.
```
$ apt list | grep libsdl2-dev
   libsdl2-dev/focal,now 2.0.10+dfsg1-3 amd64 [installed]

$ sdl2-config --cflags
   -I/usr/include/SDL2 -D_REENTRANT
```

**Comment by sr99622** [comment]
Thank you for that link, I was unaware that this was an issue with older versions of SDL2.  It seems like it has been fixed in more recent versions, 2.0.14 and 2.0.20 work with cmake.  You can work around this issue by modifying the libavio/CMakeLists.txt file.  Note that there are several CMakeLists.txt files in the project, you need the one in the libavio subdirectory.  Basically, the find_package for SDL2 is commented out, and the data from sdl2-config is modified slightly to fit in cmake format and added to target_link_libraries and target_include_directories.

```
cmake_minimum_required(VERSION 3.17)

project(libavio VERSION 1.1.0)

set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED True)
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__STDC_CONSTANT_MACROS")

list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)

if(WIN32)
    add_compile_options("/EHsc")
#    find_package(Iconv REQUIRED)
    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
    set(BUILD_SHARED_LIBS TRUE)
endif()

#find_package(SDL2 REQUIRED)
find_package(FFmpeg REQUIRED)
find_package(Qt5 COMPONENTS Widgets REQUIRED)

set(CMAKE_AUTOMOC ON)

add_library(avio SHARED
    src/Reader.cpp
    src/Exception.cpp
    src/Frame.cpp
    src/Decoder.cpp
    src/Filter.cpp
    src/Clock.cpp
    src/Display.cpp
    src/Encoder.cpp
    src/Writer.cpp
    src/Pipe.cpp
    src/GLWidget.cpp
    include/GLWidget.h
)

if (WIN32)
    target_link_libraries(avio PRIVATE
        Qt5::Widgets
        FFmpeg::FFmpeg
        #SDL2::SDL2
        SDL2
    )
else()
    target_link_libraries(avio PRIVATE
        Qt5::Widgets
        FFmpeg::FFmpeg
        #SDL2::SDL2
        SDL2
        pthread
    )
endif()

set_target_properties(avio PROPERTIES
    SOVERSION 1
)

target_include_directories(avio PUBLIC
    include
    /usr/include/SDL2
)
```

**Comment by turlapati** [comment]
The modified CMakeLists.txt resolved the problem - Thank you for prompt response...
